torch/_inductor/codegen/cpp.py

```diff
@@ -2261,6 +2261,9 @@ def select_tiling_indices():
             contig_only = (
                 contig_vars - non_contig_stride_const - non_contig_stride_other
             )
+            if len(contig_vars) == 0 and len(non_contig_stride_const) > 0:
```

Comment:
Do we have to check `len(non_contig_stride_const) > 0`?

Reply:
I am worried about the case where `contig_vars` and `non_contig_stride_const` are both empty. If this is not supposed to occur, should I add an assertion instead, like `assert len(non_contig_stride_const) > 0`?

Reply:
It might happen when the strides are not constant, e.g., with modular indexing.

Reply:
Got it. Since `non_contig_stride_other` could also be supported, I would remove len(non_contig_stride_const) > 0 then.